owncloudApp/src/main/java/com/owncloud/android/presentation/sharing/shares/PublicShareDialogFragment.kt

```diff
@@ -325,6 +338,7 @@ class PublicShareDialogFragment : DialogFragment() {
         binding.shareViaLinkPasswordValue.doOnTextChanged { text, _, _, _ ->
             capabilities?.passwordPolicy?.let { passwordPolicy ->
                 updateRequirementsPasswordPolicy(text.toString(), passwordPolicy)
+                enableCopyPasswordButton(text.toString().isNotEmpty())
```

Comment:
Why is the `copy` button activated when a character is typed in the message entry? Wouldn't it be more correct to allow copying the password when it has fulfilled the requirements? Following the same logic as the `save` button. I think it is also better in terms of UX. If I generate a password and delete some characters until the password is invalid, the copy option will still be enabled.

Reply:
I think it's a separate feature, on the one hand you can generate a password and on the other hand you can copy it. I would say copying it doesn't need to be related with the fulfilment of the policies, and in terms of UX, could be confusing for the user because "Copy" buttons attached to a text field are usually enabled when there's some non-empty text in it. What do you think @jesmrec?

Reply:
you are right that usually is enough with input characters, but this is because the most usual condition is "not empty" to avoid corner cases with non-valid strings. In this case, the condition is not only "not empty" but also "fulfilling conditions". As @Aitorbp stated, i'd only enable it when password is really useful, that means, when password passes policies. What's the aim of copying a non-valid password? we could lead users to confusion. that's what i mean.

Reply:
Ok, done 👍